src/scenarios/shared/precommands.py

```diff
@@ -184,6 +199,18 @@ def add_event_source(self, file: str, line: str, trace_statement: str):
         filepath = os.path.join(projpath, file)
         insert_after(filepath, line, trace_statement)
 
+    def install_workload(self, workloadid: str, install_args: list = ["--skip-manifest-update"]):
+        'Installs the workload, if needed'
+        if not self.has_workload:
```

Comment:
rather than requiring the user to pass in something saying the workload has already been installed, could we check it using `dotnet workload list`?

Reply:
Great find, I think that approach would work. It looks like the has-workload was added here: https://github.com/dotnet/performance/pull/2346 and may be something we can investigate modifying to be a capability check.